pkg/controller/restore_controller.go

```diff
@@ -506,48 +506,74 @@ func (c *restoreController) fetchFromBackupStorage(backupName string, pluginMana
 		return backupInfo{}, err
 	}
 
-	idx := 1
-	orderedLocations := make([]*api.BackupStorageLocation, len(locations))
-	for i := 0; i < len(locations); i++ {
-		if locations[i].Name == c.defaultBackupLocation {
-			orderedLocations[0] = locations[i]
-		} else {
-			orderedLocations[idx] = locations[i]
-			idx++
-		}
-	}
+	orderedLocations := orderedBackupLocations(locations, c.defaultBackupLocation)
 
 	logContext := c.logger.WithField("backupName", backupName)
 	var info backupInfo
 	for _, location := range orderedLocations {
-		info.objectStore, err = getObjectStoreForLocation(location, pluginManager)
-		if err != nil {
-			logContext.WithError(errors.WithStack(err)).Error("Unable to get an object store in the location ", location.Name)
-			continue
+		info, err = c.backupInfoForLocation(location, backupName, pluginManager)
+		if err == nil {
+			break
 		}
+		log := logContext.WithField("locationName", location.Name)
```

Comment:
@skriss you specifically said to put this line at the beginning of the loop. Does it make a difference? I put it here because it's close to where it's being used.